### PR TITLE
Support exit codes

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Define exit code 2 as "program incomplete, continue from checkpoint". This can be used to resubmit the job (working on that).

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Parallel/Tags/ExitCode.hpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * \brief Exit code of the executable
+ *
+ * The following exit codes are defined:
+ *
+ * - 0: Program is complete.
+ * - 1: Program aborted because of an error.
+ * - 2: Program is incomplete and should be continued from the last checkpoint.
+ *
+ * \par Design decisions
+ * - We implement the exit code as an `int` instead of an enum because its
```
Comment (nilsdeppe) [inline src/Parallel/Tags/ExitCode.hpp]: Why not have the values be hard-coded in an `enum` and then convert them to an `int` where we need them to be an `int`? I worry that the current approach requires us to match something in this documentation block. E.g.
```cpp
enum class ExitCode : int {
  Complete = 0,
  Abort = 1,
  ContinueFromCheckpoint = 2
};
```
and then
```cpp
static_cast<int>(ExitCode::ContinueFromCheckpoint)
```
If you choose to make the `enum` not an `enum class` you can just do an implicit cast.

Comment (kidder) [reply src/Parallel/Tags/ExitCode.hpp]: I would also prefer using enums in this manner...

Comment (nilsvu) [reply src/Parallel/Tags/ExitCode.hpp]: ok done!

**Review by nilsdeppe — APPROVED** [review]
Nice :D

**Pull request merged**